src/Bundler.js

```diff
@@ -332,26 +333,42 @@ class Bundler extends EventEmitter {
       let thrown = err;
 
       if (thrown.message.indexOf(`Cannot find module '${dep.name}'`) === 0) {
-        if (dep.optional) {
-          return;
-        }
-
-        thrown.message = `Cannot resolve dependency '${dep.name}'`;
-
-        // Add absolute path to the error message if the dependency specifies a relative path
-        if (dep.name.startsWith('.')) {
-          const absPath = Path.resolve(Path.dirname(asset.name), dep.name);
-          err.message += ` at '${absPath}'`;
-        }
-
-        // Generate a code frame where the dependency was used
-        if (dep.loc) {
-          await asset.loadIfNeeded();
-          thrown.loc = dep.loc;
-          thrown = asset.generateErrorMessage(thrown);
+        let isLocalFile = dep.name.startsWith('.');
+
+        if (!isLocalFile) {
+          try {
+            try {
+              logger.status(emoji.progress, `Installing ${dep.name}...`);
+              let dir = Path.dirname(asset.name);
+              await installPackage(dir, [dep.name], true, false);
+            } catch (npmError) {
+              logger.error(npmError.message);
+            }
+
+            return await this.resolveAsset(dep.name, asset.name);
+          } catch (e) {
+            if (dep.optional) {
```

Comment:
Should "optional" dependencies be installed automatically, or should we just skip these like before? Thoughts?

Reply:
By the looks of it, optional deps are pretty much dependencies that _are_ supposed to get installed/bundled, but aren’t supposed to throw an error if they aren’t able to be resolved or installed.

So ideally we should attempt to install any optional deps, but we shouldn’t block the build if the install fails.